src/hash.cr

```diff
@@ -618,17 +599,26 @@ class Hash(K, V)
   # hash.first_key? # => nil
   # ```
   def first_key?
-    @first.try &.key
+    unless empty?
+      @entries[@first].key
+    else
+      nil
+    end
   end
 
   # Returns the first value in the hash.
   def first_value
-    @first.not_nil!.value
+    nil.not_nil! if empty?
+    @entries[@first].value
   end
 
   # Similar to `#first_key?`, but returns its value.
   def first_value?
-    @first.try &.value
+    unless empty?
```

Comment:
`return nil if empty?`/`return yield if empty?` is more readable and ditto down the page.